libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -331,6 +331,13 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
 
     }
 
+    /**
+     * Used in streaming contexts to check if the streaming connection is still open for the bot to send activities.
+     */
+    public get isStreamingConnectionOpen(): boolean {
```

Comment:
would `isStreamingConnectionReady` be more clear? I'm thinking of the four states; Connected, Disconnected, Connecting, Error, specifically the case around Connecting where the connection isn't ready to send messages but it is open and waiting for a connection to come in.

Reply:
Valid point, I'm partial to the name because `"Ready"` to me implies a "just-begun" state, and I think the level of information conveyed from this `getter` is simple enough to either be "open" or "closed".

If we decide to actually dig into these four states, we can then surface this status from the server.